.github/workflows/release.yml

```
@@ -1,33 +1,63 @@
 name: Release
 
 on:
-  create:
-    tags:
-      - "v[0-9]+.[0-9]+.[0-9]+" # assumes x.y.z versioning
+  push:
+    paths: 
+      - package.json
+    branches:
+      - main
 
 jobs:
+
+  check:
```

```suggestion
  version-change-check:
```
i suggest a more informative / verbose name here